src/circleci/test/retest.clj

```diff
@@ -0,0 +1,23 @@
+(ns circleci.test.retest
+  (:require [clojure.data.xml :as xml]
+            [clojure.java.io :as io]
+            [circleci.test :as test]))
+
+(defn- failed-tests-from [root]
+  (let [suite (:name (:attrs root))]
+    (for [test-case (:content root)
+          :when (some #(= :failure (:tag %)) (:content test-case))]
+      (symbol suite (:name (:attrs test-case))))))
+
+(defn- failed-tests [report-dir]
+  (when-not (.exists (io/file report-dir))
```

Comment:
Throwing an exception here is not ideal; we really ought to just print a message and abort.

Unfortunately Clojure provides no mechanism whereby you can abort the process with a nonzero exit code when running non-interactively and yet leave the repl running when running interactively, so I don't think we're stuck choosing between two non-ideal options.